.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: GitHub Actions
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  test:
+    name: ${{ matrix.task }} (${{ matrix.os }}, ${{ matrix.compiler }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            task: r-package
+            compiler: gcc
+          - os: ubuntu-latest
+            task: r-package
+            compiler: clang
+          - os: macOS-latest
+            task: r-package
+            compiler: gcc
+          - os: macOS-latest
+            task: r-package
+            compiler: clang
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: Setup and run tests on Linux and macOS
+        if: matrix.os != 'windows-latest'
+        shell: bash
+        env:
+          COMPILER: "${{ matrix.compiler }}"
+          CONDA_ENV: "test-env"
+          GITHUB_ACTIONS: "true"
+          TASK: "${{ matrix.task }}"
```

Comment:
I think there is no reason to split environment variables definition in different places. We can move these variables from `env` step to `run` closer to other `export` commands.

Reply:
ok no problem

Reply:
changed in https://github.com/microsoft/LightGBM/pull/3119/commits/3e378bffbd908c2a75f6c95444343a9f819e9be8